### PR TITLE
feat(w3c/headers): add history header

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

File: src/w3c/headers.js [hunk 1]
```diff
@@ -697,6 +698,36 @@ export function run(conf) {
...
+    conf.historyURI = historyURL.href;
+    return;
+  }
+
+  // Do a fetch HEAD request to see if the history exists...
+  // We don't discriminate... if it's on the W3C website with a history,
+  // we show it.
+  try {
+    const response = await fetch(historyURL, { method: "HEAD" });
+    if (response.ok && response.status === 200) {
```
Comment (sidvishnoi) [inline src/w3c/headers.js:723]: `.ok` should be enough? Why test 200 specifically (and not 304 etc.)?

File: src/core/github.js [hunk 2]
```diff
@@ -153,6 +153,7 @@ export async function run(conf) {
     repoURL: ghURL.href,
     apiBase: githubAPI,
     fullName: `${org}/${repo}`,
+    commitHistoryURL: commitHistoryURL.href,
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/github.js:156]: My bad. It's used.

File: tests/spec/w3c/headers-spec.js [hunk 3]
```diff
@@ -2253,6 +2252,95 @@ describe("W3C — Headers", () => {
...
+        shortName: "test",
+        specStatus: "WD",
+        historyURI: null,
+      });
+      const doc = await makeRSDoc(ops);
+      const [history] = contains(doc, ".head dt", "History:");
+      expect(history).toBeFalsy();
+    });
+
+    it("derives the historyURI automatically when it's missing, but the document is on TR", async () => {
```
Comment (sidvishnoi) [inline tests/spec/w3c/headers-spec.js:2308]: Is this one using the fetch path?

File: tests/spec/w3c/headers-spec.js [hunk 4]
```diff
@@ -2253,6 +2252,95 @@ describe("W3C — Headers", () => {
...
+      expect(history.nextElementSibling).toBeTruthy();
+      const historyLink = history.nextElementSibling.querySelector("a");
+      expect(historyLink).toBeTruthy();
+      expect(historyLink.href).toBe(
+        "https://www.w3.org/standards/history/test"
+      );
+      expect(historyLink.textContent).toContain("Publication history");
+    });
+
+    it("includes a dd for the commit history of the document", async () => {
```
Comment (sidvishnoi) [inline tests/spec/w3c/headers-spec.js:2270]: What's historyURI in this case?

File: src/w3c/headers.js (hunk 1 above)
Comment (marcoscaceres) [reply src/w3c/headers.js:723]: yeah, agree... I was thinking of other 20X cases, but 🤷‍♂️

Comment (marcoscaceres) [inline src/w3c/headers.js:723]:
```suggestion
    if (response.ok) {
```

File: tests/spec/w3c/headers-spec.js (hunk 3 above)
Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js:2334]: not sure what you mean, but... I guess?

**Comment by marcoscaceres** [comment]
I found a significant bug... the History: should be included if either `conf.github` or `historyURI` is available.

File: tests/spec/w3c/headers-spec.js (hunk 3 above)
Comment (marcoscaceres) [reply tests/spec/w3c/headers-spec.js:2334]: Assuming you meant `response.url`, so I used that above.

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**